app/src/main/java/sw19/moring03/paint/tools/Tools.java

```diff
@@ -1,11 +1,13 @@
 package sw19.moring03.paint.tools;
 
+import android.graphics.Canvas;
+import android.graphics.Paint;
 import android.graphics.PointF;
 
 import java.util.List;
 
 public abstract class Tools {
-    public boolean draw(List<PointF> points){
-        return true;
-    }
+    protected List<PointF> points;
+    public abstract boolean draw(Canvas canvas, Paint paint);
+    public void addPoint(PointF point) {points.add(point);};
```

Comment:
remove unnecessary semi colon at the end of the function. No semicolon needed since tis is not an abstract method